src/main/java/ui/components/issuepicker/IssuePickerDialog.java

```diff
@@ -0,0 +1,210 @@
+import javafx.scene.layout.VBox;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+import ui.GuiElement;
+import ui.UI;
+import util.HTLog;
+
+/**
+ * Serves as a presenter that synchronizes changes in issues with dialog view  
+ */
+public class IssuePickerDialog extends Dialog<List<String>> {
+
+    private static final int ELEMENT_MAX_WIDTH = 400;
+    private static final Insets GROUP_PAD = new Insets(0, 0, 10, 10);
+    private static final String TITLE = "Issue Picker";
+    private static final String FXML_PATH = "fxml/IssuePickerView.fxml";
+    private static final Logger logger = HTLog.get(IssuePickerDialog.class);
+    private static final int MAX_ISSUE_SIZE = 40;
+
+    private final MultiModel models;
+    private final List<TurboIssue> allIssues;
+
+    private IssuePickerState state;
+
+    @FXML
+    private VBox issuepickerLayout;
+    @FXML
+    private VBox selectedIssues;
+    @FXML
+    private TextField issuepickerQueryField;
+    @FXML
+    private VBox suggestedIssues;
+
+
+    public IssuePickerDialog(Stage stage, MultiModel models) {
+        this.models = models;
+        this.allIssues = models.getIssues();
+
+        initUI(stage, allIssues);
+        Platform.runLater(issuepickerQueryField::requestFocus);
+    }
+
+    // UI Initialisation
+
+    @FXML
+    public void initialize() {
+        issuepickerQueryField.textProperty().addListener(
+            (observable, oldText, newText) -> handleUserInput(issuepickerQueryField.getText()));
+    }
+
+    private void initUI(Stage stage, List<TurboIssue> issues) {
+        initialiseDialog(stage);
+        setDialogPaneContent();
+        createButtons();
+
+        state = new IssuePickerState(issues, "");
+        populateSuggestedIssues(allIssues, state.getCurrentSuggestion());
+    }
+
+    private void initialiseDialog(Stage stage) {
+        initOwner(stage);
+        initModality(Modality.APPLICATION_MODAL);
+        setTitle(TITLE);
+    }
+
+    private void setDialogPaneContent() {
+        createMainLayout();
+        getDialogPane().setContent(issuepickerLayout);
+    }
+
+    /**
+     * Populates respective panes with issues that matches current user input
+     * @param state
+     */
+    private final void populatePanes(IssuePickerState state) {
+        // Population of UI elements
+        populateSelectedIssues(state.getSelectedIssues());
+        populateSuggestedIssues(state.getSuggestedIssues(), state.getCurrentSuggestion());
+
+        // Ensures dialog pane resize according to content
+        getDialogPane().getScene().getWindow().sizeToScene();
+    }
+
+    private void populateSelectedIssues(List<TurboIssue> chosenIssues) {
+        selectedIssues.getChildren().clear();
+        Map<String, FlowPane> repoContent = getRepoContent(chosenIssues);
+
+        repoContent.entrySet().forEach(entry -> {
+            selectedIssues.getChildren().addAll(createRepoTitle(entry.getKey()), entry.getValue());
+        });
```

Comment:
This isn't what I meant also. We can map createRepoTitle over the entry set to get a new collection, do then addAll on that collection. Here we're (inadvertently?) using the varargs overload, and then doing lots of `add` operations.